### ISSUE TITLE
[a11y]: FileUploader has critical violation "The tabbable element's role 'none' is not a widget role"

### ISSUE THREAD

**Issue by Carrieruan011** [carbon-design-system/carbon#12725]
### Package

carbon-components-react

### Browser

Chrome

### Operating System

Windows

### Package version

n/a

### React version

n/a

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker - Latest Deployment

### Description

The FileUploader button has a critical a11y violation "The tabbable element's role 'none' is not a widget role" that was found in our product using the IBM Accessibility Checker tool and is re-producible on the [v10](https://v7-react.carbondesignsystem.com/?path=/story/components-fileuploader--file-uploader) and [v11](https://react.carbondesignsystem.com/?path=/story/components-fileuploader--file-uploader) storybook as well.

v10 storybook

![FileUploaderCarbon](https://user-images.githubusercontent.com/118971867/203753840-d26cd6ff-a85d-48b7-a3a0-05b64c76c04a.png)
![FileUploaderCarbonRef](https://user-images.githubusercontent.com/118971867/203753853-0f4c0eb9-ad02-45ed-b5cc-c1a94fa053e3.png)

### Reproduction/example

Carbon v10 storybook - https://v7-react.carbondesignsystem.com/?path=/story/components-fileuploader--file-uploader

### Steps to reproduce

On the storybook link (either v10 or v11) for the FileUploader story, run the IBM Accessibility Checker scan. The scan should report the critical violation mentioned above (scroll to the bottom of the violations list).

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tomleah** [comment]
@tw15egan would it be possible for this to be backported this to Carbon 10, please?
